velox/functions/prestosql/tests/MapKeyExistsTest.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions {
+namespace {
+
+class MapKeyExistsTest : public test::FunctionBaseTest {};
+
+TEST_F(MapKeyExistsTest, general) {
+  // numeric key
+  auto data = makeRowVector({makeMapVectorFromJson<int64_t, int32_t>(
+      {"{1:4, 2:5, 3:6}", "{-2:5}", "{}"})});
+
+  auto result = evaluate("map_key_exists(c0, 1)", data);
+  auto expected = makeFlatVector<bool>({true, false, false});
+
+  assertEqualVectors(expected, result);
+
+  // string key
+  data = makeRowVector({makeMapVectorFromJson<std::string, std::string>({
+      "{\"ad\":null, \"bc\":null, \"cd\":null}",
+  })});
+
+  result = evaluate("map_key_exists(c0, 'bc')", data);
+  expected = makeFlatVector<bool>(true);
+
+  assertEqualVectors(expected, result);
```

Comment:
Can you write a test where value is a complex type array/another map etc?

Reply:
Could you also add tests of floating-point keys where some keys are NaN, Inf, -Inf, 0, -0, etc? And make sure the behavior matches Presto?

Reply:
Since the look up is on key and key can't be a complex type, adding tests on complex value type isn't necessary IMO.